torch/_inductor/kernel/mm_scaled_grouped.py

```diff
@@ -0,0 +1,292 @@
+                    mask=offs_bn[None, :] < n_size,
+                )
+                c = accumulator.to(tl.float32) * a_scale * b_scale
+
+                if USE_TMA_STORE:
+                    m_offset = (tile_m_idx * BLOCK_M).to(tl.int32)
+                    n_offset = (tile_n_idx * BLOCK_N).to(tl.int32)
+                    tl._experimental_descriptor_store(
+                        c_desc_ptr,
+                        c.to(c_ptr.dtype.element_ty),
+                        [m_offset, n_offset],
+                    )
+                else:
+                    idx_m = (M_start_offset + offs_am[:, None])
+                    idx_n = offs_bn[None, :]
+                    mask = offs_am[:, None] < m_size and offs_bn[None, :] < n_size
+                    {{store_output(("idx_m", "idx_n"), "c", "mask", indent_width=20)}}
+                tidx += NUM_SMS
+
+            iterated_tiles += num_tiles
+"""
+
+triton_scaled_grouped_mm_template = TritonTemplate(
+    name="scaled_grouped_mm",
+    grid=persistent_mm_grid,
+    source=triton_scaled_grouped_mm_source,
+)
+
+
+def grouped_mm_args(
+    mat1,
+    mat2,
+    layout=None,
+    out_dtype=None,
+):
+    mat1, mat2 = realize_inputs(mat1, mat2)
+    m, k1 = mat1.get_size()
+    g, k2, n = mat2.get_size()
+    k = V.graph.sizevars.guard_equals(k1, k2)
+    if layout is None:
+        from torch._inductor.ir import FixedLayout
+
+        if out_dtype is None:
+            out_dtype = mat1.get_dtype()
+
+        layout = FixedLayout(
+            mat1.get_device(),
+            out_dtype,
+            [m, n],
+        )
+    else:
+        assert out_dtype is None, "out_dtype is ignored if layout is specified."
+
+    return (g, m, n, k, layout, mat1, mat2)
+
+
+aten__scaled_grouped_mm = ExternKernelChoice(
+    torch._scaled_grouped_mm,
+    "at::_scaled_grouped_mm",
+    op_overload=aten._scaled_grouped_mm,
+    has_out_variant=False,
+)
+
+
+@register_lowering(aten._scaled_grouped_mm.default, type_promotion_kind=None)
+def tuned_scaled_grouped_mm(
+    mat_a: TensorBox,
+    mat_b: TensorBox,
+    scale_a: TensorBox,
+    scale_b: TensorBox,
+    offs: Optional[TensorBox] = None,
+    bias: Optional[TensorBox] = None,
+    scale_result: Optional[TensorBox] = None,
+    out_dtype: Optional[torch.dtype] = None,
+    use_fast_accum: bool = False,
+    layout: Optional[Layout] = None,
+) -> TensorBox:
+    g, m, n, k, layout, mat_a, mat_b = grouped_mm_args(
+        mat_a, mat_b, layout=layout, out_dtype=out_dtype
+    )
+
+    counters["aten_mm_info"][f"aten._scaled_grouped_mm.default_{g}_{m}_{n}_{k}"] += 1
+    log.info(
+        "Tuned aten._scaled_grouped_mm.default: g=%s m=%s, n=%s, k=%s, mat1_dtype=%s, mat2_dtype=%s, output_layout=%s",
+        g,
+        m,
+        n,
+        k,
+        mat_a.get_dtype(),
+        mat_b.get_dtype(),
+        layout,
+    )
+
+    device_type = get_device_type(mat_a)
+    check_supported_striding(mat_a, mat_b)
```

Comment:
should you also check that mat_a is 2d and mat_b is 3d here? aten op supports different 2d-3d combinations, but triton kernel only one